### PR TITLE
Meaningful car, boat, and bicycle proficiencies

### PR DESCRIPTION
**Pull request by gadgetpatch**
#### Summary
Features "Adds basic vehicle proficiencies for driving and boating, as well as athletic proficiencies which increase muscle engine output."

#### Purpose of change
1) Low driving skill has a negligible impact on normal gameplay, and doesn't model truly inexperienced drivers in a particularly interesting or consequential way via gameplay.
2) Muscle-powered vehicles are limited from broader use by their low output, in part due to vehicles not modelling the physical training done by athletes to improve their performance.

#### Describe the solution
2 new proficiencies were added for Driving and Boat Piloting. Operating a vehicle on land or water without the appropriate proficiency can cause the driver to either take a long time to steer, accidentally oversteer the vehicle, or fumble the controls and waste a turn instead of steering. It also limits the driver's effective driving skill when on land or water.

Drivers automatically train Driving or Boat Piloting proficiency while operating vehicles on either land or water. It takes about 4 hours for a driver to become proficient in basic vehicle controls for boats or cars. Fumbling chances are reduced, and effective driving skill increases back to normal, proportional to the driver's progress on each proficiency, as they get comfortable with the controls and handling of boats or land vehicles. Driving penalties from non-proficiency are also mitigated by high perception and dexterity stats. At Dex/Per 4, 8, and 12, steering fumble chances are 1-in-8, 1-in-10, and 1-in-12 respectively at 0% proficiency. Fumble chances further diminish by +6 at 50% proficiency, and +12 at 100%.

A new 3-tiered proficiency called Athletes's Form was added, to give players a boost when operating muscle-powered engines. Based loosely after the Archer's Form proficiencies, these model the posture and muscle groups athletes like cyclists and rowers develop to improve their performance.

Each proficiency tier gives the character a virtual +2/+4/+6 strength bonus, when calculating muscle engine power and stamina drain. This equals about 120W of extra motive power per proficiency level, allowing a master athlete to output double the foot-crank power that an untrained athlete can. Athlete's Form currently applies to all muscle engines, including foot cranks, wheelchair rims, skateboards, and rowboat oars. Implementing individual Forms for arm- and leg-powered engines was considered, but rejected as too granular.

The proficiencies automatically train while operating a muscle-powered vehicle, and take 8 hours, 16 hours, and 32 hours respectively. Intermediate athletics practice can develop Athletes Form, unlocking additional practice types for honing your Athlete's Form with workout equipment, instead of riding a bike etc.

A similar treatment was given to Archer's Form, to enable mastery training with heavy bows. Additionally, I fixed an error in the stamina drain calculation with bows; it was returning a binary 1 or 0 instead of a floating point when calculating the player strength vs bow draw weight ratio. The Archer's Form bonus now properly factors in to stamina drain reduction as well. Almost made this another pull request, but the code was in the same wheelhouse..

Lastly, proficiencies were added to appropriate hobbies and professions. A "Boating License" was added to allow players to start with boating proficiency and some swimming skill, modeled after Maine's boating safety courses. Driving and boating proficiencies were added to any profession that starts with a vehicle of that type. Intermediate and expert athletics receive Tier-1 or 2 Athletes Form as well.

#### Describe alternatives you've considered
- Adding driving penalties at 0 driving skill instead of via vehicle-specific proficiencies.
- Giving aircraft a similar treatment, with harsher consequences like total loss of control when non-proficient.
- Making existing muscle engines worse/more realistic, and adding higher-powered racing foot cranks, oars, and hand rims to compensate.
- Splitting Athlete's Form into Rower's/Lifters Form and Cyclist's/Runner's Form, and only apply them to arm-powered or leg-powered vehicles respectively.

#### Testing
Spawned Wheelchair, Bicycle, E-Bike, and Kayak. Tested each to ensure Athlete's Form training occurs only when actively moving a muscle powered vehicle, and that driving/boating proficiency training only happens when on water or land. Tested against raw stat increases, to make sure muscle-powered engines were receiving the correct virtual strength bonus from Athlete's Form. Drove vehicles long enough for growing proficiency to mitigate penalties and fumble chances. Tested and tuned non-proficient handling penalties and fumble chances at several common Dex/Per scores.

#### Additional context
BikeDemo

### CODE REVIEW

File: data/json/proficiencies/vehicles.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    "default_time_multiplier": 1.5,
+    "default_skill_penalty": 0.2,
+    "ignore_focus": true
+  },
+  {
+    "type": "proficiency",
+    "id": "prof_bike_basic",
+    "category": "prof_vehicles",
+    "name": { "str": "Cyclist's Form" },
+    "description": "You've spent some time riding bicycles and the like. When operating muscle-powered vehicles, and know how to get more power out of each stroke.",
```
Comment (cdru123) [inline data/json/proficiencies/vehicles.json:39]: In the second sentence, the "And" should be replaced with "You"

**Comment by I-am-Erk** [comment]
I think "cyclist's form" is kind of awkward wording. "Cyclist's build" might be more clear, or just "cycling"

**Comment by harakka** [comment]
I only browsed through the code cursorily, but am I correct that wheelchairing, skateboarding and rowing will also now utilize and train the cycling proficiencies?

**Comment by I-am-Erk** [comment]
> I only browsed through the code cursorily, but am I correct that wheelchairing, skateboarding and rowing will also now utilize and train the cycling proficiencies?

I think that's mentioned in the PR, it's currently anything muscle powered until they can be split into profs of their own.

File: data/json/proficiencies/athletics.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    "description": "You've honed your athletics to at least a professional level.  When it comes to making vehicles move under your own power, your body is a well-oiled machine.",
+    "can_learn": true,
+    "time_to_learn": "16 h",
+    "default_time_multiplier": 1.5,
+    "default_skill_penalty": 0.2,
+    "required_proficiencies": [ "prof_athlete_expert" ],
+    "bonuses": { "athlete": [ { "type": "strength", "value": 2 } ] },
+    "ignore_focus": true
+  }
+]
```
[the remark this answers is not in the document]
Comment (Brambor) [reply data/json/proficiencies/athletics.json:52]: You are missing a newline here, as the bot tries to tell you.

File: data/json/proficiencies/vehicles.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    "description": "You've honed your cycling to the level of a professional racer. When it comes to making vehicles move under your own power, your body is a well-oiled machine.",
+    "can_learn": true,
+    "time_to_learn": "32 h",
+    "default_time_multiplier": 1.5,
+    "default_skill_penalty": 0.2,
+    "required_proficiencies": [ "prof_bike_expert" ],
+    "bonuses": { "biking": [ { "type": "strength", "value": 2 } ] },
+    "ignore_focus": true
+  }
+]
```
[the remark this answers is not in the document]
Comment (Brambor) [reply data/json/proficiencies/vehicles.json:34]: Here too \n

[the remark this answers is not in the document]
Comment (gadgetpatch) [reply data/json/proficiencies/vehicles.json:34]: Thanks! I was just starting to suspect that's what was tripping up the formatting.

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/athletes-form-and-pulling/29533/2

**Pull request merged**